eno-core/src/test/java/fr/insee/eno/core/processing/out/steps/lunatic/LunaticFilterResultTest.java

```diff
@@ -0,0 +1,133 @@
+        tableConditionFilterType.setType(Constant.LUNATIC_LABEL_VTL);
+        tableConditionFilterType.setValue("count(CALCULATED_VARIABLE_2) < 4");
+        tableConditionFilterType.getBindingDependencies().addAll(List.of("CALCULATED_VARIABLE_2", "NUMERIC_USED_FOR_CALCULATED_VARIABLE_2"));
+        table.setConditionFilter(tableConditionFilterType);
+
+        processing = new LunaticFilterResult(enoQuestionnaire);
+    }
+
+    @Test
+    void whenApplyingShouldNotProcessComponentsWithoutFilter() {
+        input.setConditionFilter(null);
+        table.setConditionFilter(null);
+        lunaticQuestionnaire.getComponents().addAll(List.of(input, table));
+
+        processing.apply(lunaticQuestionnaire);
+        assertEquals(0, lunaticQuestionnaire.getVariables().size());
+    }
+
+    @Test
+    void whenApplyingShouldProcessSimpleComponentsWithFilter() {
+        lunaticQuestionnaire.getComponents().add(input);
+
+        processing.apply(lunaticQuestionnaire);
+        assertEquals(1, lunaticQuestionnaire.getVariables().size());
+        VariableType variable = (VariableType) lunaticQuestionnaire.getVariables().get(0);
+        testInputCalculatedVariable(variable);
+    }
+
+    @Test
+    void whenApplyingShouldProcessComplexComponentsWithFilter() {
+        lunaticQuestionnaire.getComponents().add(table);
+
+        processing.apply(lunaticQuestionnaire);
+        assertEquals(1, lunaticQuestionnaire.getVariables().size());
+        VariableType variable = (VariableType) lunaticQuestionnaire.getVariables().get(0);
+        testTableCalculatedVariable(variable);
+    }
+
+    @Test
+    void whenApplyingShouldProcessNestingComponents() {
+        Loop loop = new Loop();
+        loop.setComponentType(ComponentTypeEnum.LOOP);
+        loop.getComponents().add(input);
+
+        PairwiseLinks pairwise = new PairwiseLinks();
+        pairwise.setComponentType(ComponentTypeEnum.PAIRWISE_LINKS);
+        pairwise.getComponents().add(table);
+
+        lunaticQuestionnaire.getComponents().addAll(List.of(loop, pairwise));
+
+        processing.apply(lunaticQuestionnaire);
+        assertEquals(2, lunaticQuestionnaire.getVariables().size());
+        testInputCalculatedVariable((VariableType) lunaticQuestionnaire.getVariables().get(0));
+        testTableCalculatedVariable((VariableType) lunaticQuestionnaire.getVariables().get(1));
+    }
+
+    private void testInputCalculatedVariable(VariableType variable) {
+        assertEquals(VariableTypeEnum.CALCULATED, variable.getVariableType());
+        assertEquals("FILTER_RESULT_" + textQuestion.getName(), variable.getName());
+        assertEquals(3, variable.getBindingDependencies().size());
+        assertEquals("CALCULATED_VARIABLE", variable.getBindingDependencies().get(0));
+        assertEquals("NUMERIC1_USED_FOR_CALCULATED_VARIABLE", variable.getBindingDependencies().get(1));
+        assertEquals("NUMERIC2_USED_FOR_CALCULATED_VARIABLE", variable.getBindingDependencies().get(2));
```

Comment:
The test could be a bit more flexible since the order of variable names in binding dependencies doesn't matter (yet having these in the same order as v2 is nice, since it makes visual comparisons easier as we discussed)